### DISCUSSION TITLE
Listen to context value changes

### DISCUSSION THREAD

**Discussion by BMR11** [statelyai/xstate#5149] [Q&A]
I want to implement x-state persistence and I am following this https://stately.ai/blog/2023-10-02-persisting-state

I am using subscribe on my root-level actor  to listen to context changes:

```
  rootActor.subscribe(snapshot => {
   const updatedContext = JSON.stringify(snapshot.context);
    console.log("[Context changed]", updatedContext);
   writeToMyStorage(updatedContext)
   root});
```

It looks like this is fired many times, even if there is only a single value changed in my context object. Also, the value of `updatedContext` is ALWAYS the same (updated one but all console logs will print the same object value).

I do not want to call `writeToMyStorage` for the same `updatedContext`. I can fix this by deep comparing updatedContext from the last value but just wanted to check if I am using it wrong or if there is any better way to achieve this.

Thank you

**Comment by cmh-echomark** [comment]
@BMR11 I think you will probably want to use `getPersistedSnapshot` as the example uses:

<img width="646" alt="Screenshot 2024-12-20 at 10 26 15 AM" src="https://github.com/user-attachments/assets/6c4acadd-5294-479e-85d6-67b91f3829d3" />


Here is an entry in the docs describing the difference between `snapshot` and `getPersistedSnapshot` https://stately.ai/docs/persistence#persisting-state
